### ISSUE TITLE
Blank screen

### ISSUE THREAD

**Issue by blaustein401** [robiot/Digiexam-HackedVersion#4]
I tried using this source today and failed miserably. It simply didn't allow me to join the exam and the app aborted itself. I believe this may have been patched in one of the more recent updates. Because DigiExam is run on our own computers, though, I'm sure we could get it to work somehow. I'd be willing to pay for a proven-to-work DigiExam bypass. The issue is that we can't test it out so easily due to the lack of active exams.

@robiot and everyone else here, if you know of any possible solutions, please let me know.

**Comment by blaustein401** [comment]
I am thinking perhaps a modified virtual machine would work against DigiExam. Any ideas?

**Comment by Kardanis** [comment]
Digiexam closes other activities correctly and can still be opened in the virtual machine, but cannot join exams properly，

**Comment by robiot** [comment]
@blaustein401 @Kardanis I have another thing that doesn't prevent lock-down, but lets you start software while in lock-down.

**Comment by Kardanis** [comment]
that is enough！I can use a virtual machine to get through my boring anthropology

**Comment by blaustein401** [comment]
So how should we go about this? What is the bypass to the virtual machine prevention, and how can we start other applications whilst taking the exam?

**Comment by blaustein401** [comment]
To anyone willing to create a proven method of bypassing DigiExam protections, I will pay. The only issue is creating a demo exam with lockdown mode on. Is anyone able to make an active exam that we can verify our method on?

**Comment by robiot** [comment]
@blaustein401 @Kardanis 
I don't have time to upload source code for this right now, but this is a poc I made some months ago. This works differently.
Once you press inject, it injects the .dll file into the Digiexam process, which makes it a part of it. Therefore it will not be shut down by digiexam. Then it executes processes as SYSTEM so Digiexam can't close them. 

Remember: This is purely for educational purposes and don't use it to cheat. 

To use it with digiexam:
- Open digiexam and prepare to start the exam, but **dont do it** yet. 
- Start the digic.exe and select Digiexam in top left corner.
- Press inject
- If successful, there should now be a terminal window that appeared. This window is now a part of digiexam.
- If "Get Window Handles" returns SUCCESS you can start the exam. Else press CTRL+ALT+R to try to regenerate. If no success, restart digiexam and try again.

To try it on the lockdown example (lockdown-example.exe):
- Start lockdown example
- Open digic.exe and select Lockdown example
- Press Inject
- The terminal window should now appear
- Now you can press lockdown system (Note: This will shutdown all applications and become fullscreen, to mimic the behaviour of digiexam)
- Press alt-tab to bring the window to the foreground. Here you can see all keybinds, try them. The "Start A Webbrowser" might not work, if not, press CTRL+ALT+C to open cmd. Then open your browser from there "%LocalAppData%\Google\Chrome\Application\chrome.exe"

![image](https://user-images.githubusercontent.com/68228472/164036653-ba04069e-43c8-4aa4-9167-db41c438c777.png)
![image](https://user-images.githubusercontent.com/68228472/164036716-3f667ce2-340c-4a7b-8fd5-128497d39e17.png)

If you want to have some notes, you can put them in `C:/notes/notes.txt`. Can later be opened with CTRL+ALT+N


Extract this:
[bin1.zip](https://github.com/robiot/Digiexam-HackedVersion/files/8514155/bin1.zip)

**Comment by Kardanis** [comment]
Thank you robio，the lockdown example is sucess

**Comment by Kardanis** [comment]
By the way，this file is not available in the virtual machine

**Comment by robiot** [comment]
No, you cant use it in a virtual machine since this doesn't modify the code of digiexam.  :(

**Comment by blaustein401** [comment]
> No, you cant use it in a virtual machine since this doesn't modify the code of digiexam. :(

What about copy and pasting? Would DigiExam recognize if someone were to copy and paste text from the web onto their test?

**Comment by robiot** [comment]
> > No, you cant use it in a virtual machine since this doesn't modify the code of digiexam. :(
> 
> 
> 
> What about copy and pasting? Would DigiExam recognize if someone were to copy and paste text from the web onto their test? 

- [ ] I suppose not. As I remember it, the clipboard is only cleared once when the lockdown enables.

**Comment by blaustein401** [comment]
Okay good. And will instructors be notified if you copy and paste from the web?

**Comment by blaustein401** [comment]
Additionally, I was wondering how I'd go about creating and then accessing the notes.txt file. Much appreciate the help.

**Comment by Kardanis** [comment]
I tried it once today and found that it successfully opened web pages, but didn't show up on the screen

**Comment by robiot** [comment]
What do you mean? @Kardanis

**Comment by blaustein401** [comment]
How about notes? Did they work?

**Comment by Kardanis** [comment]
I didn't try notes, then the bin took over my page and I had to start again

**Comment by blaustein401** [comment]
I think you have to press alr+tab to switch between the exam and notes/webpage

**Comment by blaustein401** [comment]
I tried this method just now and before starting the exam, when I clicked ‘inject’ it said that it could t find the Dll. I repeatedly tried but gave up and took the test off of memory (I had studied so no big deal, my interest in Digi is more experimental than anything), but I was wondering what the issue with this was.

**Comment by robiot** [comment]
> I tried this method just now and before starting the exam, when I clicked ‘inject’ it said that it could t find the Dll. I repeatedly tried but gave up and took the test off of memory (I had studied so no big deal, my interest in Digi is more experimental than anything), but I was wondering what the issue with this was. 

Did you download the zip file and did you keep everything in the folder? Everything needs to be in the same folder, else it will not work and it will for example not find the dll.

**Comment by blaustein401** [comment]
Yes, I kept everything in the bin folder, started DigiExam, put in the exam ID, and before starting the exam I downloaded digic.exe, selected ‘DigiExam’ and clicked ‘Inject’. However, it kept on saying that the ‘Dll’ could not be found. I had the same problem on the lockdown example but after refreshing a few times it worked. On DigiExam, even after I refreshed, it didn’t.

**Comment by blaustein401** [comment]
I’m sure it’s something wrong on my part so I was wondering how I could fix this in the future.

**Comment by robiot** [comment]
> I’m sure it’s something wrong on my part so I was wondering how I could fix this in the future.

Hm I don't really know what could be the cause. I have only tested on win10, so if you use anything else, it may not work.

**Comment by blaustein401** [comment]
I am on Windows 10. Maybe I need to redownload the Zip each time I want to run the program.

**Comment by blaustein401** [comment]
I am on Windows 10. I will try again and report back to you. Maybe the problem is that each time I want to run the program I should re-download the .zip file.

**Comment by blaustein401** [comment]
Now I am not having the same issue with dll, but it can’t find window
handles. How do I solve that?

**Comment by Smurfboi007** [comment]
The example lockdown worked fine for me, haven't tested on an exam yet, but will soon. Make sure you have the Bin in your C: drive, fixed some issues for me.

**Comment by robiot** [comment]
> The example lockdown worked fine for me, haven't tested on an exam yet, but will soon. Make sure you have the Bin in your C: drive, fixed some issues for me.

Alright! Tell me how it goes, I'm curious.

**Comment by Smurfboi007** [comment]
> @blaustein401 @Kardanis I don't have time to upload source code for this right now, but this is a poc I made some months ago. This works differently. Once you press inject, it injects the .dll file into the Digiexam process, which makes it a part of it. Therefore it will not be shut down by digiexam. Then it executes processes as SYSTEM so Digiexam can't close them.
> 
> Remember: This is purely for educational purposes and don't use it to cheat.
> 
> To use it with digiexam:
> 
> * Open digiexam and prepare to start the exam, but **dont do it** yet.
> * Start the digic.exe and select Digiexam in top left corner.
> * Press inject
> * If successful, there should now be a terminal window that appeared. This window is now a part of digiexam.
> * If "Get Window Handles" returns SUCCESS you can start the exam. Else press CTRL+ALT+R to try to regenerate. If no success, restart digiexam and try again.
> 
> To try it on the lockdown example (lockdown-example.exe):
> 
> * Start lockdown example
> * Open digic.exe and select Lockdown example
> * Press Inject
> * The terminal window should now appear
> * Now you can press lockdown system (Note: This will shutdown all applications and become fullscreen, to mimic the behaviour of digiexam)
> * Press alt-tab to bring the window to the foreground. Here you can see all keybinds, try them. The "Start A Webbrowser" might not work, if not, press CTRL+ALT+C to open cmd. Then open your browser from there "%LocalAppData%\Google\Chrome\Application\chrome.exe"
> 
> ![image](https://user-images.githubusercontent.com/68228472/164036653-ba04069e-43c8-4aa4-9167-db41c438c777.png) ![image](https://user-images.githubusercontent.com/68228472/164036716-3f667ce2-340c-4a7b-8fd5-128497d39e17.png)
> 
> If you want to have some notes, you can put them in `C:/notes/notes.txt`. Can later be opened with CTRL+ALT+N
> 
> Extract this: [bin1.zip](https://github.com/robiot/Digiexam-HackedVersion/files/8514155/bin1.zip)

Could you post the source for that? It's an awesome tool! Would love to see how you made it.

**Comment by Smurfboi007** [comment]
> I tried it once today and found that it successfully opened web pages, but didn't show up on the screen

I know how to fix this. You need to do CTRL+ALT+F to toggle the fullscreen when you are there you will be able to see the webpage. Same if you have problems seeing the cmd.

**Comment by robiot** [comment]
@Smurfboi007 
> 
> Could you post the source for that? It's an awesome tool! Would love to see how you made it.
>

The source code is now pushed https://github.com/robiot/Digiexam-HackedVersion

**Comment by Smurfboi007** [comment]
Awesome man! Thank you so much for this! Will give it a try on an exam to see how it works. Been playing around with it on the example lockdown and starting to understand how to use it correctly.

**Comment by Smurfboi007** [comment]
Right, I’m back, school has started. Will try it next week on a test we have. Do you have any other updates with it or should it still work?

**Comment by Smurfboi007** [comment]
When I inject it while having digiexam open, it says sucess but no cmd shows up (I haven’t tried it on a exam yet)

**Comment by robiot** [comment]
Hm strange, Ill look into it

**Comment by Smurfboi007** [comment]
Thanks mate!

**Comment by Smurfboi007** [comment]
Did you find anything?

**Comment by Smurfboi007** [comment]
Nvm, the window now appears. Altough it says:

```
[Error] No window found. Restart the application.

[action] Get Window Handles (HWNDS) FAILURE

Iterating WINDOWS…

```
Then it keeps saying that each time i try to regenerate the window handles

**Comment by Smurfboi007** [comment]
My bad, everything works now. I updated windows and now it works like a charm. Gonna test it on Tuesday. Will be back with results!

**Comment by digixxxx** [comment]
Not to dissuade you, but Digiexam logs details about the desktop environment during the exam. For example, here the currently focused window is queried, then the name of the owning process is saved.

![image](https://user-images.githubusercontent.com/112326771/187075239-746c32e8-439d-4082-8681-ac12b6886623.png)

I don't know if teachers have access to these logs, but at the very least they're uploaded (and presumably saved) when you hand in the exam. The logging session starts when the exam starts. It doesn't automatically flag "this student is a cheater!", at least on the client side. However, after the fact, there will be documentation of you having left the exam window.

**Comment by robiot** [comment]
Where did you get/generate this code from? If it's from the executable, it's chromium's internal stuff. The whole DigiExam logic is inside the app.asar as obfuscated javascript code.

**Comment by robiot** [comment]
@Smurfboi007 
> Then it keeps saying that each time i try to regenerate the window handles

I tried it real quick and also got this error. Upon checking the task manager, it seems as if the DLL got injected into some other thread than the thread used when an exam is started.

**Comment by digixxxx** [comment]
> The whole DigiExam logic is inside the app.asar as obfuscated javascript code.

Normally, it's not possible to lock the window in fullscreen or terminate processes from a javascript context. That should tell you there's more at play. 

The app.asar package contains native code, so-called bindings, with different implementations for each platform. The code is exposed to the javascript context through a node module accessible through `require`. The screenshot was from the Windows implementation of the proprietary Digiexam security module.

**Comment by Smurfboi007** [comment]
There is another way. It requires two PCs. It's basically setting up a VNC server on your desktop or another laptop, and then leaving that one turned on at home on the day of the exam. Then just connect to it from the laptop that you bring to the exam. Regular remote desktop connections are sadly banned from Digiexam, tried that. But VNC isn't detected by Digiexam and can be used. Gonna try the method I just wrote and the DLL thing. I believe the method I wrote would eliminate the "owning process" problem since the pc at home will always have Digiexam in focus even if you click out of it.

**Comment by robiot** [comment]
> > The whole DigiExam logic is inside the app.asar as obfuscated javascript code.
> 
> 
> 
> Normally, it's not possible to lock the window in fullscreen or terminate processes from a javascript context. That should tell you there's more at play. 
> 
> 
> 
> The app.asar package contains native code, so-called bindings, with different implementations for each platform. The code is exposed to the javascript context through a node module accessible through `require`. The screenshot was from the Windows implementation of the proprietary Digiexam security module.

Was the dll you put into IDA extracted from the app.asar? Is it specifically something for digiexam, or is it a shared library imported by some dependency? Now if it is in that app.asar, that library could be easily modified to not run any of the winapi calls for terminating the desktop enviroment and other processes.

**Comment by robiot** [comment]
> There is another way. It requires two PCs. It's basically setting up a VNC server on your desktop or another laptop, and then leaving that one turned on at home on the day of the exam. Then just connect to it from the laptop that you bring to the exam. Regular remote desktop connections are sadly banned from Digiexam, tried that. But VNC isn't detected by Digiexam and can be used. Gonna try the method I just wrote and the DLL thing. I believe the method I wrote would eliminate the "owning process" problem since the pc at home will always have Digiexam in focus even if you click out of it.

Yeah that is also possible. But idk about these logs actually, and who gets access to them.

**Comment by digixxxx** [comment]
> Is it specifically something for digiexam

Yes, it's part of the dx-sec module.

> Now if it is in that app.asar, that library could be easily modified to not run any of the winapi calls for terminating the desktop enviroment and other processes.

It should be noted that tampering with the archive is also detected. The .pb file contains signatures, those are checked every time an exam is started from the internet. Meaning testing on the demo exam won't set off any alarms.

Also worth mentioning, even if you patch the calls as you suggest, the logs wouldn't check out. If you keep the fullscreen lock from being activated, the intermittent log of which window is focused would still expose you. It's also logged which processes should be terminated and which successfully are, those too would have to be spoofed with proper readings.

![image](https://user-images.githubusercontent.com/112326771/187458347-ca9d8778-91d1-4230-be3d-80964fa787c2.png)

**Comment by digixxxx** [comment]
> But VNC isn't detected by Digiexam and can be used.

Yes, the VNC test isn't very extensive. However, wouldn't whatever VNC server be killed with all other background apps? Which VNC solution are you using?

**Comment by Smurfboi007** [comment]
Found a overly conplex soloution. But it’s bassically hidding the vnc server from proccess manager, process hacker, reg entry etc. Haven’t tried with hiding it yet, but using a regular vnc server does in fact not work. Another way is putting a silent vm on your desktop and running digiexam in their and connecting a vnc server to your dekstop. This would help me for example since the laptop I have can’t handle a vm.

**Comment by digixxxx** [comment]
I don't think it's possible to hide the process in the fashion you describe. It would require injecting code in the Digiexam process, then spoofing return values related to process enumeration. 

VM sounds like a decent option. I must say VM detection is incredibly over-engineered in Digiexam, so maybe it'll prove difficult haha.

**Comment by Oliver-1-1** [comment]
You could thread attach to the process and that way have access and disable most of their calls. You could also protect the handles to your app and therefor they cant retrive a handle to your program.

**Comment by Oliver-Dahlen** [comment]
I can also give a little nerd tip.  I think Digiexam closes down dwm.exe once(i get bsod when hooked) . Protect it and draw like you are bob ross 🍡

**Comment by Smurfboi007** [comment]
> I don't think it's possible to hide the process in the fashion you describe. It would require injecting code in the Digiexam process, then spoofing return values related to process enumeration.
> 
> VM sounds like a decent option. I must say VM detection is incredibly over-engineered in Digiexam, so maybe it'll prove difficult haha.

Indeed, it was hell trying to get it undetected, used multiple VM detection software and they didn't detect it but digiexam did.

**Comment by Smurfboi007** [comment]
> You could thread attach to the process and that way have access and disable most of their calls. You could also protect the handles to your app and therefor they cant retrive a handle to your program.

That

> You could thread attach to the process and that way have access and disable most of their calls. You could also protect the handles to your app and therefor they cant retrive a handle to your program.

That does sound interesting! I have no experience in anything of what you said, could you or someone do a prototype for that? I can test it out on a real exam.

**Comment by Oliver-1-1** [comment]
> 

I already have one that protects handles. add me on discord Ol#6180

**Comment by Oliver-1-1** [comment]
> > You could thread attach to the process and that way have access and disable most of their calls. You could also protect the handles to your app and therefor they cant retrive a handle to your program.
> 
> That
> 
> > You could thread attach to the process and that way have access and disable most of their calls. You could also protect the handles to your app and therefor they cant retrive a handle to your program.
> 
> That does sound interesting! I have no experience in anything of what you said, could you or someone do a prototype for that? I can test it out on a real exam.

.

**Comment by Mach1neCoder** [comment]
> it was hell trying to get it undetected

Did you manage to get it undectected?

**Comment by Smurfboi007** [comment]
No, I did not.

**Comment by Mach1neCoder** [comment]
To verify this app dosent work currently?  I am trying to go through the code and figure a few things out.  

If you pack the app.asar it gets detected and doesn't open, regardless of if you changed something.  Seems like they might have gone ahead and fixed the issues with their program

**Comment by Smurfboi007** [comment]
Not sure what you're asking. But you shouldn't have to pack the app.asar to run digic.exe.

**Comment by Oliver-1-1** [comment]
> > it was hell trying to get it undetected
> 
> Did you manage to get it undectected?

Yes add discord Ol#6180

**Comment by Mach1neCoder** [comment]
> Not sure what you're asking. But you shouldn't have to pack the app.asar to run digic.exe.

trying to change code.  Digic.exe dosn't work on VMS 

> > > it was hell trying to get it undetected
> > 
> > 
> > Did you manage to get it undectected?
> 
> Yes add discord Ol#6180

U told me it dosnt work on VM and its payed.

**Comment by Mach1neCoder** [comment]
managed to find this in the code, does this help any of us?  Maybe by knowing what they search for, we can change system information so that it doesnt detect  @robiot 
<img width="885" alt="Screen Shot 2022-09-27 at 11 45 31 AM" src="https://user-images.githubusercontent.com/64292547/192573446-950d6a1e-51ed-4064-ac79-6e14fa3067b5.png">
<img width="735" alt="Screen Shot 2022-09-27 at 11 44 59 AM" src="https://user-images.githubusercontent.com/64292547/192573453-6ad417d1-4879-4ed8-a3df-a3dc39e64206.png">

**Comment by robiot** [comment]
@Mach1neCoder Well that code looks promising. And changing all the stuff it checks would be a possible fix. Although, I think it would be better to patch the binary to not detect anything.

**Comment by Oliver-1-1** [comment]
> > Not sure what you're asking. But you shouldn't have to pack the app.asar to run digic.exe.
> 
> trying to change code. Digic.exe dosn't work on VMS
> 
> > > > it was hell trying to get it undetected
> > > 
> > > 
> > > Did you manage to get it undectected?
> > 
> > 
> > Yes add discord Ol#6180
> 
> U told me it dosnt work on VM and its payed.

I told you it costs because i dont want to waste my time. I would not charge. But because you have a mac i cant help you

**Comment by Mach1neCoder** [comment]
Not skilled enough to do that, but as I mentioned I feel like they have a code detecting change. 

What is this btw? Is this encrypted, if so, did you ever find the encryption key?
<img width="789" alt="Screen Shot 2022-09-27 at 11 50 46 AM" src="https://user-images.githubusercontent.com/64292547/192574625-dea14eca-63fc-4129-a73e-6ec60c8363a9.png">

**Comment by Smurfboi007** [comment]
No, that is where some values are stored in your computer's memory.

**Comment by Mach1neCoder** [comment]
When you change code, I managed to get the log @Smurfboi007 @robiot 

Is this helpful to know?
```
Uncaught Exception:
SyntaxError: Error parsing /Applications/DigiExam.app/Contents/Resources/app.asar/node_modules/dx-sec/package.json: Unexpected token � in JSON at position 0
at parse (<anonymous>)
at readPackage (internal/modules/cjs/loader.js:268:20)
at readPackageExports (internal/modules/cjs/loader.js:308:15)
at applyExports (internal/modules/cjs/loader.js:450:20)
at resolveExports (internal/modules/cjs/loader.js:513:23)
at Function.Module._findPath (internal/modules/cjs/loader.js:637:31)
at Module._resolveFilename (internal/modules/cjs/loader.js:948:27)
at Function.o._resolveFilename (electron/js2c/browser_init.js:257:919)
at Module._load (internal/modules/cjs/loader.js:844:27)
at Function.Module._load (electron/js2c/asar.js:769:28)
```

**Comment by Oliver-1-1** [comment]
> When you change code, I managed to get the log @Smurfboi007 @robiot
> 
> Is this helpful to know?
> 
> ```
> Uncaught Exception:
> SyntaxError: Error parsing /Applications/DigiExam.app/Contents/Resources/app.asar/node_modules/dx-sec/package.json: Unexpected token � in JSON at position 0
> at parse (<anonymous>)
> at readPackage (internal/modules/cjs/loader.js:268:20)
> at readPackageExports (internal/modules/cjs/loader.js:308:15)
> at applyExports (internal/modules/cjs/loader.js:450:20)
> at resolveExports (internal/modules/cjs/loader.js:513:23)
> at Function.Module._findPath (internal/modules/cjs/loader.js:637:31)
> at Module._resolveFilename (internal/modules/cjs/loader.js:948:27)
> at Function.o._resolveFilename (electron/js2c/browser_init.js:257:919)
> at Module._load (internal/modules/cjs/loader.js:844:27)
> at Function.Module._load (electron/js2c/asar.js:769:28)
> ```

This just means you changed the code so it gets a runtime crash.

**Comment by Mach1neCoder** [comment]
I only changed a true to false. Does a runtime crash happen anytime code is changed

**Comment by Oliver-1-1** [comment]
> I only changed a true to false. Does a runtime crash happen anytime code is changed

Ofc not. Did you pack it again?

**Comment by Oliver-1-1** [comment]
Instead of patching their app go external. Their are plenty of ways to do it.

**Comment by Mach1neCoder** [comment]
> Ofc not. Did you pack it again?

Never unpacked it 😅. I am editing the app.asar file directly. 

No idea how to go external.

**Comment by Oliver-1-1** [comment]
> > Ofc not. Did you pack it again?
> 
> Never unpacked it 😅. I am editing the app.asar file directly.
> 
> No idea how to go external.

Just open handle and fuck em

**Comment by Mach1neCoder** [comment]
@Oliver-Dahlen  How should i go about packing without making it crash.  It crashes even if I pack the file without any changed code

**Comment by robiot** [comment]
@Mach1neCoder Are you using https://github.com/electron/asar to pack/unpack it?

**Comment by Mach1neCoder** [comment]
> @Mach1neCoder Are you using https://github.com/electron/asar to pack/unpack it?



yes installed via npm install -g asar

@robiot is that what i should be using

**Comment by Mach1neCoder** [comment]
@robiot is that what i should be using

**Comment by Mach1neCoder** [comment]
@robiot hey do you have discord?

**Comment by robiot** [comment]
> @robiot hey do you have discord? 

robiot#3455

**Comment by Mach1neCoder** [comment]
@robiot didn't work.

**Comment by Mach1neCoder** [comment]
@robiot your discord tag didn't work.  are your friend requests on?

**Comment by robiot** [comment]
@Mach1neCoder Yes, it is. Could you try to send your tag instead?

**Comment by Mach1neCoder** [comment]
@robiot Mysteriouslymythical#3936

**Comment by orixo** [comment]
> > > Ofc not. Did you pack it again?
> > 
> > 
> > Never unpacked it 😅. I am editing the app.asar file directly.
> > No idea how to go external.
> 
> Just open handle and fuck em

What is your new discord

**Comment by Chanzozo** [comment]
> [@Mach1neCoder](https://github.com/Mach1neCoder) Yes, it is. Could you try to send your tag instead?



> [@Mach1neCoder](https://github.com/Mach1neCoder) Are you using https://github.com/electron/asar to pack/unpack it?

Does the Digiexam cheat work? I haven't found a way in years, I will have Digiexam for the next 3 years, it would be very nice of you to help me, my Discord is Chanzozo

**Comment by eskile** [comment]
Hey! i know that this is a old hack but i have found myself in need for it. I used it all the time a couple years ago, but all my teachers moved on from digiexam to exam.net. Now that im in college i will bes using Digiexam again and wonder if your hack is still working, or if you know one that is working.
Vänliga hälsningar, Eskil
